### PR TITLE
USSD app with language selection

### PR DESCRIPTION
**Pull request by olpie101**
based on issue #23

### CODE REVIEW

**Comment by olpie101** [comment]
@hodgestar fix the problem. Review please

File: Gruntfile.js [hunk 1]
```diff
@@ -7,25 +7,39 @@ module.exports = function (grunt) {
...
+                    '<%= paths.src.app_ussd %>',
+                    'src/ussd/init.js'
                 ],
                 all: [
                     'src/**/*.js'
                 ]
             },
             dest: {
-                prd: 'go-app.js'
+                prd_sms: 'go-app.js',
```
Comment (hodgestar) [inline Gruntfile.js]: Perhaps let's name this `go-app-sms.js` for clarity.

File: Gruntfile.js [hunk 2]
```diff
@@ -60,15 +76,25 @@ module.exports = function (grunt) {
...
+                dest: '<%= paths.dest.prd_sms %>'
+            },
+            prd_ussd: {
+                src: ['<%= paths.src.prd_ussd %>'],
+                dest: '<%= paths.dest.prd_ussd %>'
             }
         },
 
         mochaTest: {
             test: {
```
Comment (hodgestar) [inline Gruntfile.js]: Similarly, let's call this `test_sms`.

File: src/ussd/app.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            return new EndState(name, {
+                text: 'Thanks, cheers!',
+                next: 'states:start'
+            });
+        });
+
+        self.states.add('states:select_language', function(name){
+            return new LanguageChoice(name, {
+                next: function(choice) {
+                    self.contact.extra.lang = choice.value;
```
Comment (hodgestar) [inline src/ussd/app.js]: This won't actually do anything unless you explicitly save the contact afterwards. A few things:
- The SMS app seems to save the language as `self.contact.extra.language_choice` (https://github.com/praekelt/go-usaid-mmc/blob/develop/src/app.js#L315) so perhaps we should use the same name here.
- Tests should check that the language is saved to the contact.
- Contact can be saved using:

``` javascript
self.contact.extra.language_choice = choice.value;
return
    self.im.contacts.save(self.contact)
    .then(function () {
        return "states:main_menu";
    });
```

File: test/ussd/fixtures.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+                "bar": "baz"
+            }
+        },
+        "response": {
+            "code": 200,
+            "data": {
+                "ham": "spam"
+            }
+        }
+    }];
```
Comment (hodgestar) [inline test/ussd/fixtures.js:16]: These aren't needed yet, but I think it's okay to leave them here for now as long as we remove them as soon as we add the fixtures for the real API calls.

**Comment by hodgestar** [comment]
Left a few comments, but otherwise looks good.

File: test/ussd/app.test.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        describe("when the user selects a language", function() {
+            it("should save result to contact and display the main menu", function() {
+                return tester
+                    .setup.user.state('states:select_language')
+                    .input('1')
+                    .check.interaction({
+                        state: 'states:main_menu'
+                    })
+                    .check(function(api, im, app) {
+                        assert(app.contact.extra.language_choice == 'en');
```
Comment (hodgestar) [inline test/ussd/app.test.js]: You should use

``` javascript
assert.strictEqual(app.contact.extra.language_choice, 'en');
```

it raises a nicer error message, shows the values that don't match and avoids strange things like `null == undefined` being `true`.

**Comment by hodgestar** [comment]
Left one last comment, otherwise looks good.

**Comment by hodgestar** [comment]
:+1:

**Pull request merged**